### PR TITLE
Decompile DemoEc 1 non matching

### PR DESCRIPTION
**Pull request by louist103**
Ocarina of time brought to you by...
most action funcs, setup funcs, and draw funcs named

### CODE REVIEW

**Comment by fig02** [comment]
skelanime updated and MANY cleanups implemented.
Louis, the file probably looks very different now lol. Have a scroll through to see the kinds of things I changed.

The way the array of function pointers for init/update/draw are being accessed feels hella fake but I couldnt improve it.

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_Demo_Ec/z_demo_ec.c [hunk 1]
```diff
@@ -9,7 +10,1333 @@ void DemoEc_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+    DemoEc_UpdateSkelAnime(this);
+    func_8096D594(this, globalCtx);
+    DemoEc_UpdateBgFlags(this, globalCtx);
+}
+
+void DemoEc_DrawWindmillMan(DemoEc* this, GlobalContext* globalCtx) {
+    DemoEc_DrawSkeleton(this, globalCtx, &D_06005F20, &D_06006920, NULL, NULL);
+}
+
+void DemEc_InitKokiriBoy(DemoEc* this, GlobalContext* globalCtx) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Ec/z_demo_ec.c:423]:
```suggestion
void DemoEc_InitKokiriBoy(DemoEc* this, GlobalContext* globalCtx) {
```

File: src/overlays/actors/ovl_Demo_Ec/z_demo_ec.c [hunk 2]
```diff
@@ -9,7 +10,1333 @@ void DemoEc_Destroy(Actor* thisx, GlobalContext* globalCtx);
...
+
+    if ((secondaryBankIndex < 0) || (primaryBankIndex < 0)) {
+        // Demo_Ec_main_bank: Bank unreadable arg_data =%
+        osSyncPrintf(VT_FGCOL(RED) "Demo_Ec_main_bank:バンクを読めない arg_data = %d!\n" VT_RST, type);
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    if ((Object_IsLoaded(&globalCtx->objectCtx, primaryBankIndex)) &&
+        (Object_IsLoaded(&globalCtx->objectCtx, secondaryBankIndex))) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Ec/z_demo_ec.c:1245]:
```suggestion
    if (Object_IsLoaded(&globalCtx->objectCtx, primaryBankIndex) &&
        Object_IsLoaded(&globalCtx->objectCtx, secondaryBankIndex)) {
```

File: undefined_syms.txt [hunk 3]
```diff
@@ -696,7 +759,7 @@ D_0600B4E4 = 0x0600B4E4;
 D_0600BE4C = 0x0600BE4C;
 D_06001978 = 0x06001978;
 D_0600A324 = 0x0600A324;
-D_0600A99C = 0x0600A99C;
+D_0600A99C = 0x0600A99C;
```
Comment (Roman971) [inline undefined_syms.txt:762]:
```suggestion
D_0600A99C = 0x0600A99C;
```

File: undefined_syms.txt [hunk 4]
```diff
@@ -683,6 +683,69 @@ D_060135EC = 0x060135EC;
...
+D_06005AC0 = 0x06005AC0;
+D_06005990 = 0x06005990;
+D_06005880 = 0x06005880;
+D_06009198 = 0x06009198;
+D_06009430 = 0x06009430;
+D_06009690 = 0x06009690;
+D_060035D8 = 0x060035D8;
+D_060074C8 = 0x060074C8;
+D_0600DE80 = 0x0600DE80;
+D_06003770 = 0x06003770;
```
Comment (Roman971) [inline undefined_syms.txt:748]:
```suggestion
D_06003770 = 0x06003770;

```

File: src/overlays/actors/ovl_Demo_Ec/z_demo_ec.c (hunk 1 above)
Comment (fig02) [reply src/overlays/actors/ovl_Demo_Ec/z_demo_ec.c:423]: sorry what is this change showing it looks the same to me

Comment (fig02) [reply src/overlays/actors/ovl_Demo_Ec/z_demo_ec.c:423]: nvm

File: undefined_syms.txt (hunk 4 above)
Comment (fig02) [reply undefined_syms.txt:748]: theres two sections for en_test anyway so i just removed the duplicate

**Review by Roman971 — APPROVED** [review]

**Pull request merged**